modules/README.md

```diff
@@ -146,4 +149,53 @@ interface ISafeProtocol712SignatureValidator {
 }
 ```
 
+- Signature Validator Manager
+
+
+```solidity
+interface ISafeProtocolSignatureValidatorManager {
+    /**
+     * @param dataHash Hash of the data that is signed
+     * @param data Arbitrary data containing the following layout:
+     *             Layout of the data:
+     *              0x00 - 0x04: selector
+     *              0x04 - 0x36: domainSeparator
+     *              0x36 - 0x68: typeHash
+     *              0x68 - 0x6C: encodedData length
+     *              0x6C - 0x6C + encodedData length: encodedData
+     *              0x6C + encodedData length - 0x6C + encodedData length + 0x20: payload length
+     *              0x6C + encodedData length + 0x20 - end: payload
+     * @return bytes4 Value returned by the Signature Validator Contract
+     */
+    function isSignatureValid(bytes32 dataHash, bytes data) return (bytes4 magicValue);
+}
+
+```
+
+### Sequence diagram for Signature Validation
+
+The diagram below illustrates the sequence of calls that are made when a signature is validated. The `Account` sets the `SignatureValidatorContract` via the `SignatureValidatorManager`. When a signature for an account is to be validated by an external entity, here referred as `ExternalContract`, the `ExternalContract` contract calls the `SignatureValidatorManager` which checks if the `SignatureValidatorContract` is listed and not flagged. If the contract is listed and not flagged, the `SignatureValidatorManager` calls the `SignatureValidatorContract` to check if the signature is valid.
+
+```mermaid
+sequenceDiagram
+	participant Account
+	participant ExternalContract
+	participant SignatureValidatorManager
+	participant RegistryContract
+	participant SignatureValidatorContract
+
+	Account->>SignatureValidatorManager: Set SignatureValidatorContract
+		SignatureValidatorManager->>RegistryContract: Check if contract is listed and not flagged
```

Comment:
contract is the signature validator here, right?

Reply:
Here its the `SignatureValidatorManager `. 
`SignatureValidatorManager` will check if there exists a `SignatureValidator` for given domain, if yes, it will further check in `Registry` if `SignatureValidator` is listed and not flagged.

I will update the specs